Assets/Scripts/ROS Scripts/Camera Streaming Scripts/ImageVisualizer.cs

```diff
@@ -0,0 +1,45 @@
+﻿using ROSBridgeLib.sensor_msgs;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ImageVisualizer : MonoBehaviour {
+
+    //attached gameobject with texture2D that will change per frame
+    //GameObject s;
+    Texture2D originalTexture;
+
+    string encoding;
+    uint width;
+    uint height;
+    ImageMsg image;
+    RawImage m_RawImage;
+
+
+    // Use this for initialization
+    void Start () {
+      
+
+    }
+
+    // Update is called once per frame
+    void Update () {
+		
+	}
+
+	public void SetFrame(ImageMsg data, string videoPlayer)
+    {
+        Debug.Log(videoPlayer);
+        if (videoPlayer == "")
+        {
+            return;
+        }
+        Texture2D tex = new Texture2D((int)data.GetWidth(), (int)data.GetHeight(), TextureFormat.RGB24, false);
+        GameObject.Find(videoPlayer).GetComponent<RawImage>().enabled = true;
```

Comment:
Where is ImageVisualizer attached? whats the reason why we predefine videoPlayer to have only a few fixed ones rather than dynamically generate it somehow (writing this question I realize that somehow might not be that easy)?

Reply:
We can dynamically generate it, its just predefined for now until we have a good idea on what the finalized players will look like.